server/controllers/events.js

```diff
@@ -1,61 +1,107 @@
+      const events = await Event.find()
+        .populate("user", "displayName")
+        .lean()
+        .exec();
+
       // return all events
       res.json(events);
     } catch (err) {
-      console.log(err);
+      next(err);
     }
   },
-  getOne: async (req, res) => {
+  getOne: async (req, res, next) => {
     try {
-      const event = await Event.findById(req.params.id);
+      const { id } = req.params;
+
+      // Check if the ID is valid
+      if (!mongoose.isValidObjectId(id)) {
+        throw httpError(404);
+      }
+
+      // Get event by id
+      const event = await Event.findById(id).lean().exec();
+
+      // Check if event exists
+      if (!event) {
+        throw httpError(404);
+      }
+
       res.json(event);
     } catch (err) {
-      console.log(err);
+      next(err);
     }
   },
-  deleteEvent: async (req, res) => {
+  deleteEvent: async (req, res, next) => {
     try {
-      const eventId = req.params.id;
-      //checks if an event exists that _id, user, and req.user._id match. This is to prevent users that are authenticated from deleting events they do not author.
-      const event = await Event.findOne({ _id: eventId, user: req.user._id });
-      if (!event) { return res.status(401).send({ message: 'You are not the author of this event' }); }
-      await Event.deleteOne({ _id: eventId });
-      res.json({ message: 'Event deleted' });
-    } catch (error) {
-      console.error(error);
-      res.send(500);
+      const { id } = req.params;
+
+      // Check if the ID is valid
+      if (!mongoose.isValidObjectId(id)) {
+        throw httpError(404);
+      }
+
+      // Prevent users that are authenticated from deleting events they do not author.
+      const event = await Event.findOne({ _id: id, user: req.user._id });
+      if (!event) {
+        throw httpError(401);
+      }
+
+      // Delete event by id
+      await Event.findByIdAndDelete(id);
+
+      res.sendStatus(204);
+    } catch (err) {
+      next(err);
     }
   },
-  deleteAllEvents: async (req, res) => {
+  deleteAllEvents: async (req, res, next) => {
     try {
-      const groupId = req.params.groupId;
-      //checks if an event exists that _id, user, and req.user._id match. This is to prevent users that are authenticated from deleting events they do not author.
-      const event = await Event.findOne({ groupId: groupId, user: req.user._id })
-      if (!event) { return res.status(401).send({ message: 'You are not the author of this event' }); }
-      await Event.deleteMany({ groupId: groupId });
-      res.json({ message: 'Events deleted' });
-    } catch (error) {
-      console.error(error);
-      res.send(500);
+      const { groupId } = req.params;
+
+      // Prevent users that are authenticated from deleting events they do not author.
+      const count = await Event.countDocuments({
+        groupId,
+        user: req.user._id,
+      }).exec();
+
+      if (count === 0) {
+        throw httpError(401);
```

Comment:
I think that a 404 here would be more appropriate.

> The HyperText Transfer Protocol (HTTP) 401 Unauthorized response status code indicates that the client request has not been completed because it lacks valid authentication credentials for the requested resource.

It's not that they lack valid authentication; this route's purpose is "delete all events in this group", which, if the group doesn't exist, then they're trying to delete something that doesn't exist.  Not found!

Reply:
I've updated it to 404.
This is from the document I linked in another comment:
<img width="1059" alt="image" src="https://user-images.githubusercontent.com/25715951/212154259-26ee9a34-2d05-4927-a294-422750890f60.png">

Do you think it's a good idea to remove the query that counts the documents and only leave the `Event.deleteMany` query?

Reply:
Yeah - we can basically trust `deleteMany` to delete all of the 

And re the status code - I think that source misunderstand what, exactly, is meant to be idempotent.  This stackoverflow answer explains it pretty well, I think: https://stackoverflow.com/questions/4088350/is-rest-delete-really-idempotent